### PR TITLE
[Reputation Oracle] Implemented Cron and Payout Modules

### PR DESCRIPTION
**Pull request by eugenvoronov**
## Description
Improved structure of the project. Crone and Payout modules was implemented.

## Summary of changes
- Implemented cron job database locks as implemented in Job Launcher.
- Created cron job modules and move cron jobs into it.
- Created a Payout module to handle payouts. Use payout service from cron jobs service.
- Improved Reputation Service.
- Implemented unit tests for each error case.

## How test the changes
`yarn test`

## Related issues
Reputation Oracle - Cron jobs refactor
[#1508](https://github.com/humanprotocol/human-protocol/issues/1508)

### CODE REVIEW

File: packages/apps/reputation-oracle/server/src/modules/reputation/reputation.service.ts [hunk 1]
```diff
@@ -1,44 +1,166 @@
...
+    } else {
+      await this.decreaseReputation(
+        chainId,
+        recordingOracleAddress,
+        ReputationEntityType.RECORDING_ORACLE,
+      );
     }
 
-    return ReputationLevel.MEDIUM;
+    return;
```
Comment (portuu3) [inline packages/apps/reputation-oracle/server/src/modules/reputation/reputation.service.ts:153]: can remove this

File: packages/apps/reputation-oracle/server/src/modules/reputation/reputation.service.ts [hunk 2]
```diff
@@ -1,44 +1,166 @@
...
+  /**
+   * Perform reputation assessment based on the completion status of a job and its associated entities.
+   * Retrieves necessary data from the escrow client, including manifest and final results URLs,
+   * and delegates reputation adjustments to specialized methods.
+   * @param chainId The ID of the blockchain chain.
+   * @param escrowAddress The address of the escrow contract.
+   * @param checkPassed A boolean indicating if the job completion check passed.
+   * @returns {Promise<void>} A Promise indicating the completion of reputation assessment.
+   */
+  public async doAssessReputationScores(
```
Comment (portuu3) [inline packages/apps/reputation-oracle/server/src/modules/reputation/reputation.service.ts:46]: Here we could do something like [this](https://github.com/humanprotocol/human-protocol/pull/1574/files), so we have a specific assessment for each job type.

File: packages/apps/reputation-oracle/server/src/modules/payout/payout.service.ts [hunk 3]
```diff
@@ -0,0 +1,227 @@
...
+
+  /**
+   * Calculates processing results based on the manifest type.
+   * It retrieves the signer, builds an escrow client, retrieves the manifest URL,
+   * downloads the manifest data, and processes it accordingly.
+   * @param chainId The ID of the blockchain chain.
+   * @param escrowAddress The address of the escrow contract.
+   * @returns {Promise<ProcessingResultDto>} An object containing processing results including recipients, amounts, and storage data.
+   */
+  public async calculateResults(
```
Comment (portuu3) [inline packages/apps/reputation-oracle/server/src/modules/payout/payout.service.ts:80]: Here we can do something like [this](https://github.com/humanprotocol/human-protocol/pull/1574/files)

File: packages/apps/reputation-oracle/server/src/modules/payout/payout.dto.ts [hunk 4]
```diff
@@ -0,0 +1,11 @@
+export class PayoutsResultDto {
+  /**
+   * URL to the stored results.
+   */
+  url: string;
+
+  /**
+   * Boolean indicating if the checks passed.
+   */
+  checkPassed: boolean;
```
Comment (portuu3) [inline packages/apps/reputation-oracle/server/src/modules/payout/payout.dto.ts:10]: what is checkPassed for?

**Review by portuu3 — CHANGES_REQUESTED** [review]

File: packages/apps/reputation-oracle/server/src/modules/payout/payout.dto.ts (hunk 4 above)
Comment (eugenvoronov) [reply packages/apps/reputation-oracle/server/src/modules/payout/payout.dto.ts:10]: If checkPassed = false we will decrease reputation

**Review by portuu3 — APPROVED** [review]

**Review by portuu3 — APPROVED** [review]

**Pull request merged**